fbw-a380x/src/wasm/systems/a380_systems/src/air_conditioning/local_controllers/trim_air_drive_device.rs

```diff
@@ -100,22 +95,20 @@ impl<const ZONES: usize, const ENGINES: usize> TrimAirDriveDevice<ZONES, ENGINES
     }
 
     fn fault_determination(&mut self) {
-        self.fault = match self.active_channel.has_fault() {
-            true => {
-                if self.stand_by_channel.has_fault() {
-                    Some(TaddFault::BothChannelsFault)
-                } else {
-                    self.switch_active_channel();
-                    Some(TaddFault::OneChannelFault)
-                }
-            }
-            false => {
-                if self.stand_by_channel.has_fault() {
-                    Some(TaddFault::OneChannelFault)
-                } else {
-                    None
-                }
+        self.active_channel.update_fault();
+        self.stand_by_channel.update_fault();
+
+        self.fault = if self.active_channel.has_fault() {
+            if self.stand_by_channel.has_fault() {
+                Some(TaddFault::BothChannelsFault)
+            } else {
+                self.switch_active_channel();
+                Some(TaddFault::OneChannelFault)
             }
+        } else if self.stand_by_channel.has_fault() {
+            Some(TaddFault::OneChannelFault)
+        } else {
+            None
         };
```

Comment:
Another possibility (don't need to accept):
```suggestion
        self.fault = match (
            self.active_channel.has_fault(),
            self.stand_by_channel.has_fault(),
        ) {
            (true, true) => Some(TaddFault::BothChannelsFault),
            (false, false) => None,
            (ac, _) => {
                if ac {
                    self.switch_active_channel();
                }
                Some(TaddFault::OneChannelFault)
            }
        };
```

Reply:
I love this suggestion, thanks! I've applied it to all the local controllers that use the same `fault_determination` function. This could probably be further abstracted with a macro but I'll leave it for a future refactor 😄.